src/main/resources/com/cloudbees/jenkins/support/SupportAction/index.jelly

```diff
@@ -60,42 +60,31 @@
-                none
-              </div>
               <button class="jenkins-button jenkins-button--primary" onclick="document.getElementById('categories_dialog').close();return false;">
                 ${it.close}
               </button>
+              <button id="copyToClipboardButton" class="jenkins-button jenkins-submit-button jenkins-button--primary" style="margin-left:1em;" onclick="copyToClipboard(); return false;">
+                ${it.genericComponentCopy}
+              </button>
             </f:section>
           </dialog>
           <f:entry title="${it.chooseGeneralComponent}" field="mapText" help="/plugin/support-core/help/help-chooseGeneralComponent.html">
             <select id="idMapSelect" class="jenkins-select__input select">
-              <option value="1F79CB2F69F57">${it.genericComponentDefault}</option>
-              <option value="1F79CBAF69FD7">${it.genericComponentConfigurationFiles}</option>
-              <option value="1F7FCBBFEFFD7">${it.genericComponentPerformanceData}</option>
+              <option value="H4sIAAAAAAAAE5VU0W7bMAz8mT0O+ofV6bAVcZHFad8Vi3XYSKRGUnH794OdYtiewj5Jgu544umkkUsYM7d0BNDwCnRG0qCtVhYLWGoOQ44X6LiUSGmwaKiGo34dbzM3rdT7t2V9iMcMHsojJ+iZ0FhcEncNc/rVoLmq7xsR0rSSXOW/HbnZw3XHhU8FCdUkGl4+1ciW4+quC7wTfnvvmF5waosU0w2WWhRrNQynZoln6rhUJiDz8q6jm5Z5FvjdQC0Mmef9dX44CcS0pEK9ldZ2n2qKBh2Qgfi8nCaQPlKcfISPG97y5LL/EPXsxa4JuhOe1XeUFf/kBC+4jpvPxYfnfic8gurwrgalBxMctWMyINMvfVSnv1f+f/nzsQ0LSNhATJnH80cqPhWFe7qgMBUge46Cy7fiuobvmGEDOgpWY9liQZfcI9jMcv65JO8ljj6tPVxAFNZH6rdzJ1xBDH0a/zylW56vyNRKDT9wOi3/zF/H/wARjOM1/gUAAA==">${it.genericComponentDefault}</option>
+              <option value="H4sIAAAAAAAAE6WUy3LbMAxFf6bLDv8hkdNpM1bqWk72NAnLqEmCAUAr+fuO5D5XhqcrUcN7eQbPQNmFRC3uAcR9h3LCIk5arcTqMNfkhuTP0FHOvsRBvaIoBvkYrjtXLdeHt/l/5/cJLJYnitBTQSU2Ie4bpvitQTO9vm2lYBkXk+n5uz01fbzcmPQxY0FR9opnayCBygHHAyYQ1y3nT5jmjFcqUNTCXZNfanML6qsegf/w5Cbghunt/WJuc7BUrrhEPWurbjg2jTQVK00STQyvDUTdkGjaXs5m+y/s5fuf1N2Rwce5rW/L1nONXqGDosC2co4jcO+LH22GrolSXtNo6tKfDW2VL41iFe+8nKzaZbrumSaxRbnon43iWddRsxXo8aXfMAUQGd5FIfegjEE6KgpF5UPvxVi6i/+fybC5FTOwW4GPicLpxjZf0A/ljEwlQ9EXzzivXFMZ5ulfgQTGqsRrzGjCPYFOxKcvc1MffLCxtnAGFljWhz2dG6YKrGhj/DWl13K+KGPL1X3G8Thv0d8Z/wFM9Gf4GgcAAA==">${it.genericComponentConfigurationFiles}</option>
+              <option value="H4sIAAAAAAAAE6WV3XIaMQyFXyaXGb9DsqGlHUgpP7k3XrGo2JYryUDevuOlKe0VYnq13vE5862PZG2g5EKk2m8BxP2AfMAsTmopxOowlehW0R+ho5R87lfqFUUxyGO47XypqUzO7X3ttxEsllfqYXKGUJXYxGiGOWW06p8rxv57hWr6nGXNGfMwmuSRePiQBXQl1qEZTsSHXaSTC0VcV2S9Z/B9O/tTUKT8sMACETNcNyzopy1V/XrZMen7hBlF2SserZEEyjscdhhBXDeuP2FsxS6UIauFOyM/tsU9qG+6B77y5C7ggun8fjHXdljKN1yinrUWt9pX7emUrTSJdGL4WUHUrSKdlpe12f6BvTz/k3ptnvvS2pTeK3SQFdhWzmEAnvvsB5uhq6KUZjSYuvR3Q1vlY6NYxWsvB6v2c2dVjvfwmekktjxG/cYobrqOqq2UU/BlI36AKYrSwD6ZEn+bL5gCiKzeRSHNQRmDdJQVssrD3IuxNS7+f26eza2YgN0L+D5SONx5jUb0JB+RKSfI+uYZ29/EVLw2XV5AAmNR4hkmNOFeQdtU/9Iuzc4HG2sJR2CBcTzZ41wwFWBFG+OvKXAr81HZ11TcFId9m9J/Ev8FWTBcDfUHAAA=">${it.genericComponentPerformanceData}</option>
```

Comment:
This is not maintainable. Define a Java method to construct these hashes by some means.

Reply:
Yep, this is another excellent idea. I modified this part as well. Now, we can override the `getDefautlPreChooseOptions` method in the component to choose the predefinied options where the component will be included:

```
@Override
    public SupportAction.PreChooseOptions[] getDefautlPreChooseOptions() {
        return new SupportAction.PreChooseOptions[]{ SupportAction.PreChooseOptions.Default, SupportAction.PreChooseOptions.ConfigurationFiles, SupportAction.PreChooseOptions.PerformanceData };
    }
```